### PR TITLE
record progress on bubble choice parent after completing its sublevel

### PR DESCRIPTION
**Pull request by davidsbailey**
## Background

* finishes [PLAT-121](https://codedotorg.atlassian.net/browse/PLAT-121)

## Description

in order to efficiently query progress for bubble choice levels, we want to store progress on the bubble choice level, rather than only on its sublevels.

## Testing story

Added a unit test to verify progress is being recorded in both places.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/activities_controller.rb [hunk 1]
```diff
@@ -184,9 +184,28 @@ def track_progress_for_user
         level_source_id: @level_source.try(:id),
         pairing_user_ids: pairing_user_ids,
       )
+
+      is_sublevel = !@script_level.levels.include?(@level)
```
Comment (clareconstantine) [inline dashboard/app/controllers/activities_controller.rb:188]: Just to clarify my understanding - for bubble choice, we create a ScriptLevel for the parent level, for which `@script_level.levels` returns an array containing the parent level. Do the sublevels also have their own ScriptLevels?

**Review by clareconstantine — APPROVED** [review]
looks good, just one question about the structure behind this!

File: dashboard/app/controllers/activities_controller.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/controllers/activities_controller.rb:188]: nope -- the hierarchy works like this:
* ScriptLevel contains 1 Level (or multiple levels, if using "level swapping")
* that Level can contain sublevels (if it is a `BubbleChoice` or `LevelGroup`)
* sublevels do not have their own `ScriptLevel`s, but
* sublevels do have their own `UserLevel`s (indexed on script + level + user)

it's not ideal... but it's better than it used to be :)

Comment (clareconstantine) [reply dashboard/app/controllers/activities_controller.rb:188]: That clears up my confusion. Thank you! :)

**Pull request merged**